### ISSUE TITLE
Support Indexed Types

### ISSUE THREAD

**Issue by william-lohan** [angular/vscode-ng-language-service#110]
Can indexers be supported?
example:

```
@Component({
  selector: 'component-selector',
  template: '<div>{{ indexer.foo }}</div>' // red squiggly under `indexer.foo`
})
export class MyComponent {

  public indexer: { [x: string]: string } = {
    foo: 'bar'
  };

}
```

**Comment by gatimus** [comment]
Any update on this? I also tried `<div>{{ indexer.['foo'] }}</div>` but get:
> Parser Error: Unexpected token [

**Comment by chuckjaz** [comment]
The current model for validation uses a custom type checker. I intended to enhance this type checker in `ngc` as well. Unfortunately, it didn't hold up well and making it support everything it needed to was too costly. For `ngc` I switched to generating TypeScript that matches the semantics of the expression and having TypeScript type check it. This has yet to make it to the language service, however. When it does this will be supported.
